01-front-end-fundamentals/git-ssh/readme.md

```diff
@@ -0,0 +1,161 @@
+
+Then in Github:
+
+1. In the top right corner of any page, click your  profile photo, then click Settings.
+2. In the user settings sidebar, click SSH keys.
+3. Click Add SSH key.
+4. In the Title field, add a descriptive label for the key.
+5. Paste your key into the "Key" field
+6. Click Add key
+7. Confirm the action by entering your GitHub password
+
+## Step 5: Checking your Connection
+To make sure everything is working, you'll now try to SSH into GitHub. When you do this, you will be asked to authenticate this action using your password, which is the SSH key passphrase you created earlier.
+
+To do this let's:
+
+1. Open Terminal and enter:
+
+  ```
+  $ ssh -T git@github.com
+  # Attempts to ssh to GitHub
+  ```
+
+2. You may see this warning:
+
+  ```
+  The authenticity of host 'github.com (207.97.227.239)' can't be established.
+  # RSA key fingerprint is 16:27:ac:a5:76:28:2d:36:63:1b:56:4d:eb:df:a6:48.
+  # Are you sure you want to continue connecting (yes/no)?
+  ```
+
+Verify the fingerprint in the message you see matches the following message, then type yes
+
+3. You should then see output similar to this:
+  ```
+  Hi username! You've successfully authenticated, but GitHub does not
+  # provide shell access.
+  ```
+If the username in the message is yours, you've successfully set up your SSH key!
+
+*Note*: If you If you receive a message about "access denied," please notify an instructor or you can [read these instructions for diagnosing the issue](https://help.github.com/articles/error-permission-denied-publickey/)
+
+*Note*: If you're switching from HTTPS to SSH, you'll now need to update your remote repository URLs. For more information, [see Changing a remote's URL](https://help.github.com/articles/changing-a-remote-s-url/).
+
+## Setup Upstream for Curriculum Repo
+Since we have all already forked and cloned the GA-DC/Curriculum repository onto our local machine - we need a way to be able to update our local copy with any changes being made to the master repository.
+
+One of the ways we can do this is by establishing a link to this remote repository so that we can get all the changes anytime we want.
+
+The convention is to add an `upstream` remote.
+
+This remote is where we can pull from at any time, but do not have permission to push our changes.
+
+So let's go ahead and set up our new remote for our local curriculum repository.
+
+## YOU-DO: Setup Upstream for Curriculum Repo
+1. Students should visit the [Curriculum](https://github.com/ga-dc/curriculum)'s page and copy the ssh clone url to the clipboard
+2. In our terminal, change directories into our local curriculum directory
+```
+# in ~/wdi
+$ cd curriculum
+```
+3. See all current remotes with:
+`$ git remote -v`
+4. Add a new remote named upstream to our repository
+5. Pull down updates from the upstream remote
+
+When you finish, please go back into Github and make sure your profile has your accurate FULL NAME, and a relevant Photo of you.
```

Comment:
Your presence on github is important! Potential employers will visit your profile!